### PR TITLE
Logging developer documentation 

### PR DESCRIPTION
**Pull request by sbesson**
This PR adds a page to the Sphinx developer documentation to describe the logging framework (slf4j + log4j/logback) used by Bio-Formats and the methods of the `DebugTools` utility class.
It also includes some documentation of the breaking logging API changes in https://github.com/openmicroscopy/bioformats/pull/2372.

### CODE REVIEW

File: docs/sphinx/developers/logging.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
+Logging
+=======
+
+Logging is the process of writing log messages during the execution.
```
Comment (mtbc) [inline docs/sphinx/developers/logging.txt]: maybe -"the"

File: docs/sphinx/developers/logging.txt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  or via a prior call to ``DebugTools.enableLogging()``.
+
+- ``DebugTools.enableLogging(level)`` will initialize the logging framework
+  under the same conditions as described above and set the root logger level if
+  the initialization was succesful.
+
+- ``DebugTools.setRootLevel(level)`` will override the level of the root logger
+  independently of how the logging system was initialized.
+
+- ``DebugTools.enableIJLogging()`` (logback-only) will add an ImageJ specific
```
Comment (mtbc) [inline docs/sphinx/developers/logging.txt]: hyphenate ImageJ-specific

**Comment by jburel** [comment]
``Two underlying logging framework`` =>``frameworks``

File: docs/sphinx/developers/logging.txt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+series of framework-agnostic methods for the initialization and control of the
+logging system. This class uses reflection to detect the underlying logging
+framework and delegate the method calls to either
+:javadoc:`Log4jTools <loci/common/Log4jTools.html>` or
+:javadoc:`LogbackTools <loci/common/LogbackTools.html>`.
+
+The main methods are described below:
+
+- ``DebugTools.enableLogging()`` will initialize the underlying logging 
+  framework. This call will result in a no-op if the logging has been
```
Comment (simleo) [inline docs/sphinx/developers/logging.txt]: Remove "the"

File: docs/sphinx/developers/logging.txt [hunk 4]
```diff
@@ -0,0 +1,51 @@
+Logging
+=======
+
+Logging is the process of writing log messages during execution.
```
Comment (simleo) [inline docs/sphinx/developers/logging.txt]: This is tautological and not very informative for a software developer. I'd remove lines 3-7 altogether and start directly with specific info about logging in BF

**Comment by simleo** [comment]
Docs build with no errors. All links in the added/changed sections work fine. Re: content, I've added two minor inline notes, otherwise fine to merge.

File: docs/sphinx/developers/logging.txt (hunk 4 above)
Comment (sbesson) [inline docs/sphinx/developers/logging.txt]: A single introductory sentence was certainly either too much or not enough. I will remove it as I don't have the capacity to write an authoritative introduction about logging.

**Comment by sbesson** [comment]
Pushed

**Comment by simleo** [comment]
Latest changes look good. OK to merge.

**Pull request merged**